app/Community/RouteServiceProvider.php

```diff
@@ -228,11 +229,10 @@ protected function mapWebRoutes(): void
 
                 //     // Route::get('history', [PlayerHistoryController::class, 'index'])->name('history.index');
 
-                //     /*
-                //      * inbox
-                //      */
-                //     Route::resource('message', MessageController::class)->except('index', 'edit', 'update');
-                //     Route::resource('messages', MessageController::class)->only('index')->names(['index' => 'message.index']);
+                /*
+                 * messages
+                 */
+                Route::resource('messages', MessageThreadsController::class)->except(['edit', 'update']);
```

Comment:
I see the previous duality of plural for list/non-plural for items. Do you want me to switch back to that?

Reply:
just realized that those routes aren't guarded yet; should read like this:
```php
Route::group([
    'middleware' => ['auth'], // TODO add 'verified' middleware
], function () {
    Route::resource('messages', MessageThreadController::class)->only('index')->names(['index' => 'message.index']);
    Route::get('message/create', [MessageThreadController::class, 'create'])->name('message.create');
    Route::resource('message-thread', MessageThreadController::class)->except('index', 'create', 'edit', 'update');
});
```

Reply:
> I'd still go with separating controllers for messages and message threads.

I'm still struggling with this decision. The user doesn't care about the separation of the models. Why wouldn't everything be `message-thread`?

```
messages.index -> message-thread.index
messages.show -> message-thread.show
messages.create -> message-thread.create
messages.store -> message-thread.store
messages.destroy -> message-thread.destroy
```

And
```
Route::resource('message-threads', MessageThreadController::class)->only('index')->names(['index' => 'message-thread.index']);
Route::resource('message-thread', MessageThreadController::class)->except('index', 'edit', 'update');
```

Personally, I like the shorter URLS: `/messages` and `/message/id`, but I realize they're effectively for the threads.

Reply:
As part of separating the parts of the controller, I settled on being able to create/store `messages`, and show/destory `message_threads`. `index` is associated to `message_threads`, but pathed through `messages`:
```
/messages                        message-thread.index

/message-thread/{message_thread} message-thread.show
/message-thread/{message_thread} message-thread.destroy

/message/create                  message.create
/message                         message.store
```
I'm still uncertain on the `message.store` route, as it creates a message and appends it to a thread. It's not really updating a `message` or a `message_thread`.

Reply:
This looks really good now imo! Thanks for adjusting it like this 🙏 

`message.store` is fine like this I think as it's affecting both resources and it would have to work like this either way. Having a store route that matches the create view looks good to me.

Reply:
I'm still not sure it's appropriate in either place. If we leave it here, we'll have to move it later if we give users the ability to edit previous messages (not something that's usually allowed anyway). If we move it to `message_thread.store`, then we lose a similar ability to edit the message thread title.

Reply:
for editing `edit` and `update` routes are used. `create` and `store` are only for creation.